src/app/resources/view-resources/resources-view.component.ts

```diff
@@ -89,6 +102,66 @@ export class ResourcesViewComponent implements OnInit, OnDestroy {
       this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
     }
     this.couchSrc = this.urlPrefix + resource._id + '/' + filename;
+    // resource rating
+    this.couchService.get('ratings/_all_docs?include_docs=true')
+      .subscribe((data) => {
+        this.ratings = data.rows.map(ratings => {
+          return ratings.doc;
+        }).filter(rt  => {
+          return rt['type'] === 'resource' && rt['item'] === resource._id;
+        });
+        let rate_sum = 0;
+        let has_rated = 0;
+        let total_rating = 0;
+        let male_rating = 0;
+        let female_rating = 0;
+        this.ratings.map(rate => {
+          has_rated = (rate.user === this.userService.get().name) ? rate.rate : has_rated;
+          total_rating++;
+          (rate.gender === 'M') ? male_rating++ : female_rating++ ;

```

Comment:
A few problems here:

- The gender will be in `rate.user.gender`, so this will always increment `female_rating`
- The value being stored is `'male'` or '`female`', so it will never equal `'M'`
- We should explicitly check that it equals '`female'` before incrementing `female_rating` in case the user has not filled out their profile.
- Using the ternary operator in this case is not ideal.  It's not intended to be a shorthand if/else with code running in each case, but rather to decide between values to assign to a variable based on a conditional.  This is a good case to use a `switch`, as it makes it clear to someone reading the code that we are running a different block of code based on the value of `rate.user.gender`